src/test/java/core/basesyntax/ActionStrategyTest.java

```diff
@@ -0,0 +1,86 @@
+package core.basesyntax;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import core.basesyntax.service.ActionStrategy;
+import core.basesyntax.service.ActionStrategyImpl;
+import core.basesyntax.service.handler.ActionHandler;
+import core.basesyntax.service.handler.BalanceActionHandler;
+import core.basesyntax.service.handler.PurchaseActionHandler;
+import core.basesyntax.service.handler.ReturnActionHandler;
+import core.basesyntax.service.handler.SupplyActionHandler;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ActionStrategyTest {
+    private static final String BALANCE_ACTION_KEY = "b";
+    private static final String SUPPLY_ACTION_KEY = "s";
+    private static final String PURCHASE_ACTION_KEY = "p";
+    private static final String RETURN_ACTION_KEY = "r";
+    private static final String INVALID_ACTION_KEY = "i";
+    private static ActionStrategy actionStrategy;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        Map<String, ActionHandler> actionHandlerMap = new HashMap<>();
```

Comment:
you can have private methods in tests, FYI

Reply:
added private method